hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOMRatisSnapshots.java

```diff
@@ -1001,6 +1027,230 @@ public void testInstallCorruptedCheckpointFailure() throws Exception {
+    int lastIndex = currentCompactionLogPath.lastIndexOf(OM_KEY_PREFIX);
+    String compactionLogsPath = currentCompactionLogPath
+        .substring(0, lastIndex);
+    int numberOfLogFiles = 0;
+    long contentLength;
+    Path compactionLogPath = Paths.get(compactionLogsPath);
+    Path currentCompactionLog = Paths.get(currentCompactionLogPath);
+    try (BufferedReader bufferedReader =
+             Files.newBufferedReader(currentCompactionLog);
+         DirectoryStream<Path> files =
+             Files.newDirectoryStream(compactionLogPath)) {
+      contentLength = bufferedReader.lines()
+          .mapToLong(String::length)
+          .reduce(0L, Long::sum);
+      for (Path ignored : files) {
+        numberOfLogFiles++;
+      }
+    }
+
+    // Check whether newly created snapshot gets processed by SFS
+    newKeys = writeKeys(1);
+    SnapshotInfo newSnapshot = createOzoneSnapshot(newLeaderOM,
+        snapshotName + RandomStringUtils.randomNumeric(5));
+    Assertions.assertNotNull(newSnapshot);
+    File omMetadataDir =
+        OMStorage.getOmDbDir(newLeaderOM.getConfiguration());
+    String snapshotDir = omMetadataDir + OM_KEY_PREFIX + OM_SNAPSHOT_DIR;
+    Path filePath =
+        Paths.get(snapshotDir + OM_KEY_PREFIX + FILTERED_SNAPSHOTS);
+    Assertions.assertTrue(Files.exists(filePath));
+    GenericTestUtils.waitFor(() -> {
+      List<String> processedSnapshotIds;
+      try {
+        processedSnapshotIds = Files.readAllLines(filePath);
+      } catch (IOException e) {
+        throw new RuntimeException(e);
+      }
+
+      return processedSnapshotIds.contains(newSnapshot.getSnapshotId()
+          .toString());
+    }, 1000, 30000);
+
+    // Check whether newly created snapshot data can be reclaimed
+    client.getObjectStore()
+        .deleteSnapshot(volumeName, bucketName, newSnapshot.getName());
+    GenericTestUtils.waitFor(() -> {
+      Table<String, SnapshotInfo> snapshotInfoTable =
+          newLeaderOM.getMetadataManager().getSnapshotInfoTable();
+      try {
+        return null == snapshotInfoTable.get(newSnapshot.getTableKey());
+      } catch (IOException e) {
+        throw new RuntimeException(e);
+      }
+    }, 1000, 10000);
+
+    // Check whether newly created keys data can be reclaimed
+    Table<String, OmKeyInfo> omKeyInfoTableBeforeDeletion = newLeaderOM
+        .getMetadataManager()
+        .getKeyTable(ozoneBucket.getBucketLayout());
+    String newKey = OM_KEY_PREFIX + ozoneBucket.getVolumeName() +
+        OM_KEY_PREFIX + ozoneBucket.getName() +
+        OM_KEY_PREFIX + newKeys.get(0);
+    Assertions.assertNotNull(omKeyInfoTableBeforeDeletion.get(newKey));
+    ozoneBucket.deleteKeys(newKeys);
+    GenericTestUtils.waitFor(() -> {
+      Table<String, OmKeyInfo> omKeyInfoTableAfterDeletion = newLeaderOM
+          .getMetadataManager()
+          .getKeyTable(ozoneBucket.getBucketLayout());
+      try {
+        return Objects.isNull(omKeyInfoTableAfterDeletion.get(newKey));
```

Comment:
This not the preferred test for reclamation.  The key gets removed from the keytable immediatly, (and gets moved to the deleted table.)  The real test is to check if the blocks corresponding to the key have been removed from the datanode.  That might be hard to do.  Please look into it.